crates/red_knot_vendored/vendor/typeshed/stdlib/red_knot.pyi

```diff
@@ -0,0 +1,25 @@
+from typing import TypeVarTuple
+
+Ts = TypeVarTuple("Ts")
+
+# Operations on types
+class Negate[T]: ...
+class Intersection[*Ts]: ...
+class ClassLiteral[T]: ...
+
+# Predicates on types
+class _IsEquivalentTo[S, T]: ...
+class _IsSubtypeOf[S, T]: ...
+class _IsAssignableTo[S, T]: ...
+class _IsDisjointFrom[S, T]: ...
+class _IsFullyStatic[T]: ...
+
+def is_equivalent_to[S, T](x: S, y: T) -> _IsEquivalentTo[S, T]: ...
```

Comment:
These functions would probably be good examples for https://discuss.python.org/t/pep-747-typeexpr-type-hint-for-a-type-expression/55984